Added EnchancedSideBar

ronelliott opened the pull request.

titoBouzout commented:
> Added EnchancedSideBar #5851

This user does not have my permission to redistribute my own packages on package control in any kind of form.

And you Ron Elliott from The Control Group San Diego, C , you are a piece of shit, bully you can go to hell. Fuck you idiot.

titoBouzout commented:
To be clear, nobody has my permission to distribute any of my package on this channel.  And specially about this package that is someone popular, the only person that may or may not decide to disturbute it is on this thread https://github.com/titoBouzout/SideBarEnhancements/issues/332

mottosso commented:
Actually, he does, according to your own license.

- https://github.com/titoBouzout/SideBarEnhancements#license (GPL)

The relevant portion is here.

> This program is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation. http://www.gnu.org/licenses/gpl.html

titoBouzout commented:
Yes sure, but just because you can do something does not mean you should do it. There are codes, he came with treats. I'm willing to allow people but not n this terms. He came to ask with bad attitude I replied, clarified and expressed he does not have my consent to do this, he came anyways. enough

wbond commented:
@titoBouzout The author has forked your code in a way that is legal and following the spirit of the license you chose. Now you are asking me to not include his project (different name, derived code) because you don't like it that someone else wants to distribute the software? The **whole** point of the GPL is that you are ensuring the end users can always use your code, and that no one can restrict it.

I respect that you don't want to distribute your code via Package Control's default channel because you don't like our policies. If you want to relicense your code in such a way that redistribution is not allowed, you can limit forks from taking new versions, but please don't ask Package Control to enforce more strict licenses on your code than *you* are.

----

@ronelliott As it stands now, your readme is kind of a mess. You haven't modified the one from tito's repo, so it contains all kinds of incorrect information. Additionally, the tagged version you have and your changelog are at odds. Your version is `0.1.0`, whereas the changelog mentions version `2.*`.

titoBouzout commented:
... I know is legal, what I mean is that because is just legal does not
mean you should do it. I'm expressly asking to not merge this request. He
forked it fine, merging in has nothing to do with the license. Looks to me
thid user does not even reviewed the code, tomorrow he wipes a hard drive
and what he gonna say is gpl tito wrote it and I fork it. He is an
irresponsible bully that has nothing better todo, he asked I denied it.

On Mon, Oct 3, 2016, 12:00 Will Bond <notifications@github.com> wrote:

> @titoBouzout <https://github.com/titoBouzout> The author has forked your
> code in a way that is legal and following the spirit of the license you
> chose. Now you are asking me to not include his project (different name,
> derived code) because you don't like it that someone else wants to
> distribute the software? The *whole* point of the GPL is that you are
> ensuring the end users can always use your code, and that no one can
> restrict it.
>
> I respect that you don't want to distribute your code via Package
> Control's default channel because you don't like our policies. If you want
> to relicense your code in such a way that redistribution is not allowed,
> you can limit forks from taking new versions, but please don't ask Package
> Control to enforce more strict licenses on your code than *you* are.
> ------------------------------
>
> @ronelliott <https://github.com/ronelliott> As it stands now, your readme
> is kind of a mess. You haven't modified the one from tito's repo, so it
> contains all kinds of incorrect information. Additionally, the tagged
> version you have and your changelog are at odds. Your version is 0.1.0,
> whereas the changelog mentions version 2.*.
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/wbond/package_control_channel/pull/5851#issuecomment-251130314>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAD6nFUYu_ZbtVhlvBLFqdpxB3WqC9LEks5qwRiTgaJpZM4KMBz1>
> .
>
-- 

__
Tito

wbond commented:
I understand that – what you are asking me to do is become a mediator of disagreements, which isn't a job I am willing to do. I make sure people follow the technical policies, and try to encourage them to not duplicate packages or use the name "Sublime" in their packages, but if someone really wants their package in, I'd rather just say yes and move on. This is a thankless job, and honestly mostly just takes away from time spent with my family.

I think the cleanest way for you to handle it is to change your license so that users are free to use your code, but not redistribute it. Otherwise you are asking me to give you special treatment.

ronelliott commented:
@wbond I'll get the README updated ASAP. Are we all good and in agreement on the license here?

ronelliott commented:
@wbond README updated.

ronelliott commented:
Also to be clear here, there is no duplicate package being publish with this PR. If there is another package with the same features I will happily remove this package. I even chose a name different from the original package to allow the original author to return to Package Control at a future date. I'm not sure how much more of proper procedure I can follow here, and if I have not followed proper procedure please point it out.

This fork and PR was created because the author of the original work removed their work due to the semver changes and no longer wished to publish to Package Control. This PR is currently just replacing the work in Package Control. Going forward I will maintain and improve upon this project.

As far as me removing history, I've no plans to nor am I event sure why I would remove it. That said, I did "destroy" history when renaming the readme.md file to README.md to follow current conventions. Those changes can be found in these two commits:

[https://github.com/ronelliott/EnhancedSideBar/commit/08ae45248ca4cf73d35c1d3dc302f616d9270fe1](https://github.com/ronelliott/EnhancedSideBar/commit/08ae45248ca4cf73d35c1d3dc302f616d9270fe1)

[https://github.com/ronelliott/EnhancedSideBar/commit/24855f2d6d17fc3701b1ac5449a23bc4e82c20e4](https://github.com/ronelliott/EnhancedSideBar/commit/24855f2d6d17fc3701b1ac5449a23bc4e82c20e4)

As far as the "no code review" comment. I forked the project, cloned it and symlinked it into my Packages folder, verified it's functionality, and proceeded with the PR with the intention to completely rewrite the plugin at a future date (decided after a cursory glance over the work). I have since updated all broken file references (though some would / may call this omission a feature, as it allowed seamless functionality with the previous SideBarEnhancements package), updated the readme to point all references to the proper repository, removed any unnecessary or extraneous entries in the readme, updated the screenshot in the readme and made a couple of minor style changes for some code sections.

FichteFoll commented:
Looks fine by me. Maybe a reference to the original package could be added, although it shows as a "fork" already.

Since this is quite *controversial*, I'll let @wbond review this as well.

titoBouzout commented:
https://github.com/wbond/package_control_channel/pull/5858

FichteFoll commented:
With #5858 merged, I assume we can close this now, right, @ronelliott?

ronelliott commented:
works for me :)
